File: src/trees/BinaryNode.java
package trees;

public class BinaryNode {

    public Finisher element;
    public BinaryNode left;
    public BinaryNode right;

    public BinaryNode(Finisher finisher) {
        this.element = finisher;
        this.left = null;
        this.right = null;
    }
}

File: src-python/8-Trees/BinaryNode.py
# necessary before python 3.14 to use type hints for enclosing classes
from __future__ import annotations
from Finisher import Finisher

class BinaryNode:

    def __init__(self, finisher: Finisher):
        self.element = finisher
        self.left: BinaryNode = None
        self.right: BinaryNode = None